wgengine/packet/packet.go

```diff
@@ -6,27 +6,30 @@ package packet
 
 import (
 	"encoding/binary"
+	"errors"
 	"fmt"
-	"log"
 	"net"
 	"strings"
 
 	"tailscale.com/types/strbuilder"
 )
 
+// IPProto encodes an IP protocol, such as TCP or UDP.
 type IPProto int
 
 const (
+	// Junk marks packets that appear malformed
+	// or that have an IP protocol which we do not know how to parse.
 	Junk IPProto = iota
+	// IPv6 marks IPv6 packets. Although it is not an IP protocol,
+	// it is used instead of Junk for added clarity.
+	IPv6
```

Comment:
I assume this is informational for now?

Instead of this, could we add an IPFamily to QDecode and set it to IPv4/IPv6?

Reply:
That's probably a bad idea. Our packet filtering should be completely separated between IPv4 and IPv6. A packet with IPProto=IPv6 is functionally equivalent to type Junk except for the log message.

There is no IPv6 support at all right now. Adding it is very nontrivial.